calico-vpp-agent/connectivity/ipsec.go

```diff
@@ -122,7 +122,7 @@ func (p *IpsecProvider) RescanState() {
 
 		p.log.Infof("Using async workers for ipsec, nbDataThread=%d", p.nDataThreads)
 		for i := 0; i < p.nDataThreads; i++ {
-			err = p.vpp.SetCryptoWorker(uint32(i), false)
+			err = p.vpp.SetCryptoWorker(uint32(i), false) // FIXME is this correct? It seems like disabling instead of enabling worker threads for IPSec
```

Comment:
Agreed, this is quite unclear. The nDataThreads is set by `FetchNDataThreads()` in `common.go` but we probably should do a better job at naming it.

The idea when using IPSEC, is that we use the first nDataThread workers for RX/TXing packets, and the rest for async crypto. So we need to disable async crypto on the first "DataThreads"

Reply:
Thanks, yes I think this makes sense. It should be clearer in the context of ipsec